allennlp/models/simple_tagger.py

```diff
@@ -39,23 +55,43 @@ class SimpleTagger(Model):
     def __init__(self, vocab: Vocabulary,
                  text_field_embedder: TextFieldEmbedder,
                  encoder: Seq2SeqEncoder,
+                 calculate_span_f1: bool = None,
+                 label_encoding: Optional[str] = None,
+                 label_namespace: str = "labels",
+                 verbose_metrics: bool = False,
                  initializer: InitializerApplicator = InitializerApplicator(),
                  regularizer: Optional[RegularizerApplicator] = None) -> None:
         super(SimpleTagger, self).__init__(vocab, regularizer)
 
+        self.label_namespace = label_namespace
         self.text_field_embedder = text_field_embedder
-        self.num_classes = self.vocab.get_vocab_size("labels")
+        self.num_classes = self.vocab.get_vocab_size(label_namespace)
         self.encoder = encoder
+        self._verbose_metrics = verbose_metrics
         self.tag_projection_layer = TimeDistributed(Linear(self.encoder.get_output_dim(),
                                                            self.num_classes))
 
         check_dimensions_match(text_field_embedder.get_output_dim(), encoder.get_input_dim(),
                                "text field embedding dim", "encoder input dim")
+
+        if calculate_span_f1 is None:
+            calculate_span_f1 = label_encoding is not None
```

Comment:
as I understand this, if you specify a label encoding, calculate_span_f1 is implicitly true, and if you don't then it's implicitly false? if that's the case, then why have a calculate_span_f1 parameter at all?

Reply:
🤦‍♂️ sorry, was just copy-pasting from CRF tagger. If we remove it, not sure that the names `label_encoding` and `label_namespace` together suggest that they're used for calculating span F1---seems prudent to rename them? Any suggestions? I was thinking `span_f1_label_encoding` and `span_f1_label_namespace`, but maybe that's a bit verbose?

Reply:
I guess the difference over there is that `label_encoding` does something useful even if `calculate_span_f1` is false, whereas here it doesn't.

I guess it's probably for the best to keep the APIs of these classes as similar as possible

Reply:
but I would add a comment here explaining the reasoning, because it's not obvious